lib/extensions/ar_adapter/ar_checkin_checkout_callbacks.rb

```diff
@@ -0,0 +1,22 @@
+ActiveRecord::ConnectionAdapters::AbstractAdapter.class_eval do
+  set_callback(:checkout, :after, :log_after_checkout)
+  set_callback(:checkin,  :after, :log_after_checkin)
+
+  def connection_info_for_logging
+    pool             = ActiveRecord::Base.connection_pool
+    pool_size        = pool.size
+    count            = pool.connections.count
+    in_use           = pool.connections.count(&:in_use?)
+    waiting_in_queue = pool.num_waiting_in_queue
+
+    "pool_size: #{pool_size}, connections: #{count}, in use: #{in_use}, waiting_in_queue: #{waiting_in_queue}"
+  end
+
+  def log_after_checkin
+    _log.debug { connection_info_for_logging } if ActiveRecord::Base.connected?
```

Comment:
Note, I chose our logger because Rails logging is debug in dev mode and this logging might be noisy in that normal workflow.  Also, it felt like more application specific than framework related.